File: app/models/actb/ox_mark_info.rb
module Actb
  class OxMarkInfo
    include ApplicationMemoryRecord
    memory_record [
      { key: "correct", name: "正解",     score: 1,  sound_key: :correct, delay_second: 0.7, },
      { key: "timeout", name: "時間切れ", score: 0,  sound_key: :timeout, delay_second: 0.7, },
      { key: "mistake", name: "不正解",   score: -1, sound_key: :mistake, delay_second: 0.5, },
    ]
  end
end

File: front_app/components/Wkbk/models/ox_mark_info.js
import MemoryRecord from 'js-memory-record'

export class OxMarkInfo extends MemoryRecord {
  static get define() {
    return [
      { key: "correct", name: "正解",   sound_key: "o", name: "正解", icon: "circle-outline", type: "is-danger",  },
      { key: "mistake", name: "不正解", sound_key: "x", name: "誤答", icon: "close",          type: "is-success", },
    ]
  }

  get icon_attrs() {
    return { icon: this.icon, type: this.type }
  }
}
